src/brpc/stream.cpp

```diff
@@ -462,8 +462,12 @@ class MessageBatcher {
     ~MessageBatcher() { flush(); }
     void flush() {
         if (_size > 0 && _s->_options.handler != NULL) {
-            _s->_options.handler->on_received_messages(
+            int ret = _s->_options.handler->on_received_messages(
                     _s->id(), _storage, _size);
+            if (BAIDU_UNLIKELY(ret != 0)) {
+                LOG(WARNING) << "Fail to receive message for stream: " << _s->id();
+                _s->_options.handler->on_failure(_s->id());
```

Comment:
I think it's unnecessary to call `on_failure` here. Since user knows what will return from `on_received_messages`, they can call `on_failure` in `on_received_messages`.